pkg/dexidp/provider.go

```diff
@@ -98,6 +123,38 @@ func (p *dexProvider) Configure(ctx context.Context, req provider.ConfigureReque
 	resp.ResourceData = client
 }
 
+func constructDexClient(host string, tlsCfg *tlsConfiguration) (api.DexClient, error) {
+	if tlsCfg == nil {
+		return client.New(host, insecure.NewCredentials())
+	}
+
+	caCrt := tlsCfg.ServerCrt.ValueString()
+	clientCrt := tlsCfg.ClientCrt.ValueString()
+	clientKey := tlsCfg.ClientKey.ValueString()
+
+	if (clientCrt == "" && clientKey != "") || (clientCrt != "" && clientKey == "") {
+		return nil, fmt.Errorf("client crt and key must both be set or both be empty")
+	}
+
+	cPool := x509.NewCertPool()
+
+	if !cPool.AppendCertsFromPEM([]byte(caCrt)) {
+		return nil, fmt.Errorf("failed to parse CA crt")
+	}
+
+	clientCert, err := tls.X509KeyPair([]byte(clientCrt), []byte(clientKey))
+	if err != nil {
+		return nil, fmt.Errorf("failed to load client cert/key pair: %v", err)
+	}
+
+	clientTLSConfig := &tls.Config{
+		RootCAs:      cPool,
+		Certificates: []tls.Certificate{clientCert},
+	}
```

Comment:
Probably move this TLSConfig part to the client package.

So this package just has to do something like

```shell
mTLSConfig, err := client.NewMTLSConfig(ca, cert, key)
client.New(host, credentials.NewTLS(mTLSConfig))
```